common/geth/multihoming_client.go

```diff
@@ -0,0 +1,546 @@
+package geth
+
+import (
+	"context"
+	"math/big"
+
+	dacommon "github.com/Layr-Labs/eigenda/common"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type MultiHomingClient struct {
+	RPCs       []*EthClient
+	rpcUrls    []string
+	NumRetries int
+	Logger     logging.Logger
+	*FailoverController
+}
+
+var _ dacommon.EthClient = (*MultiHomingClient)(nil)
+
+func NewMultiHomingClient(config EthClientConfig, logger logging.Logger) (*MultiHomingClient, error) {
```

Comment:
It would be helpful to document the high level behavior of this client. For example,
```
// NewMultiHomingClient is an EthClient that automatically handles RPC failures and retries by cycling through
// multiple RPC clients... <when does it retry with a different client? any caveats?>
```